lang/src/org/partiql/lang/eval/CompileOptions.kt

```diff
@@ -95,7 +97,7 @@ data class CompileOptions private constructor (
         fun undefinedVariable(value: UndefinedVariableBehavior) = set { copy(undefinedVariable = value) }
         fun projectionIteration(value: ProjectionIterationBehavior) = set { copy(projectionIteration = value) }
         fun rewriterMode(value: RewritingMode) = set { copy(rewritingMode = value) }
-
+        fun thunkOptions(value: (Throwable, SourceLocationMeta?) -> Nothing) = set { copy(thunkOptions = ThunkOptions(value))}
```

Comment:
In looking at this now I think we need to introduce a [`typealias`](https://kotlinlang.org/docs/reference/type-aliases.html):

```Kotlin
typealias ThunkExceptionHandler = (Throwable, SourceLocationMeta?) -> Nothing)
```

Then use the name `ThunkExceptionHandler` any place `(Throwable, SourceLocationMeta?) -> Nothing)` is used.

Also, I think the type of the `value` here should be `ThunkOptions`.

Furthermore, for consistency, I think `ThunkOptions` require builders following the pattern as `CompileOptions` and `EvaluationSession`.

Reply:
Okay. I will incorporate these.

Reply:
I have made these changes.